### PR TITLE
Add Helm3 support to bake action on KubernetesManifestV0

### LINKED ISSUE #12511: KubernetesManifestV0 returns Error: unknown flag: --name
## Required Information

**Question, Bug, or Feature?**  
*Type*: Bug

**Enter Task Name**: KubernetesManifestV0

## Environment
Server - Azure Pipelines
Agent - Hosted (ubuntu-16.04)

## Issue Description

The `bake` action of the task started failing with the following error:
```
##[error]Error: unknown flag: --name
```
I checked and the installed version of Helm and it is v3.1.1 somehow. The KubernetesManifestV0 task does not support this version yet and tries to include the `--name` argument for the release name, but the argument does not exist anymore.

### Task logs

``` 
##[debug]   template
##[debug]   /home/vsts/work/1/registration/registration-v0.1.1.tgz
##[debug]   --name
##[debug]   registration
##[debug]   --namespace
##[debug]   sso
##[debug]   --set
##[debug]   image.tag=[redacted]
##[debug]   --set
##[debug]   appsettings.resourceProxy.baseAddress=[redacted]
##[debug]   --set
##[debug]   appsettings.identityProvider.authority=[redacted]
##[debug]   --set
##[debug]   appsettings.identityProvider.validIssuers[0]=[redacted]
##[debug]   --set
##[debug]   appsettings.identityProvider.validIssuers[1]=[redacted]
##[debug]   --set
##[debug]   appsettings.azureKeyVault.keyVaultEndpoint=***
##[debug]   --set
##[debug]   appsettings.azureKeyVault.clientId=***
##[debug]   --set
##[debug]   secretValues.azureKeyVault.clientSecret=***
##[debug]   --set
##[debug]   istio.hosts[0]=[redacted]
##[debug]   --set
##[debug]   istio.hosts[1]=[redacted]
##[debug]   --set
##[debug]   appsettings.serilog.properties.Region=[redacted]
##[debug]   --set
##[debug]   appsettings.serilog.properties.ClusterName=[redacted]
##[debug]   --set
##[debug]   appsettings.serilog.properties.ImageTag=[redacted]
##[debug]task result: Failed
##[error]Error: unknown flag: --name

##[debug]Processed: ##vso[task.issue type=error;]Error: unknown flag: --name%0A
##[debug]Processed: ##vso[task.complete result=Failed;]Error: unknown flag: --name%0A
Finishing: Bake registration
```

### Error logs

```
##[error]Error: unknown flag: --name
```

### PR DESCRIPTION
**Pull request by colindembovsky**
Adding support for helm3 rendering of templates using `bake` action. Fixes #12511 

[Proposed Label: Deployment]

### CODE REVIEW

**Comment by colindembovsky** [comment]
@ds-ms - done! Initially I couldn't see a clean way of testing the helm version, so that's why I went with user input. This is much better.

File: Tasks/KubernetesManifestV0/src/actions/bake.ts [hunk 1]
```diff
@@ -109,7 +109,7 @@ export async function bake(ignoreSslErrors?: boolean) {
     const renderType = tl.getInput('renderType', true);
     let renderEngine: RenderEngine;
     switch (renderType) {
-        case 'helm2':
+        case 'helm':
```
Comment (ds-ms) [inline Tasks/KubernetesManifestV0/src/actions/bake.ts]: You'll need to. add `case :  helm2' as well for back compatibility

Comment (colindembovsky) [reply Tasks/KubernetesManifestV0/src/actions/bake.ts]: Good catch!

File: Tasks/Common/kubernetes-common-v2/helm-object-model.ts [hunk 2]
```diff
@@ -9,24 +9,44 @@ export interface NameValuePair {
...
         this.helmPath = kubectlPath;
+        this.checkHelmVersion();
         if (!!namespace) {
             this.namespace = namespace;
         } else {
             this.namespace = 'default';
         }
     }
 
+    private checkHelmVersion() {
```
Comment (ds-ms) [inline Tasks/Common/kubernetes-common-v2/helm-object-model.ts]: `setHelmVersion` maybe

File: Tasks/KubernetesManifestV0/Tests/L0.ts [hunk 3]
```diff
@@ -232,12 +267,52 @@ describe('Kubernetes Manifests Suite', function () {
...
+    it('Run should succeed with helm3 bake should override values with : correctly', (done: MochaDone) => {
+        const tp = path.join(__dirname, 'TestSetup.js');
+        const tr: ttm.MockTestRunner = new ttm.MockTestRunner(tp);
+        process.env[shared.TestEnvVars.action] = shared.Actions.bake;
+        process.env[shared.TestEnvVars.helmChart] = 'helmChart';
+        process.env[shared.TestEnvVars.renderType] = 'helm';
+        process.env[shared.TestEnvVars.helmVersion] = "v3";
         process.env[shared.TestEnvVars.overrides] = 'name:value:with:colons';
         process.env.RemoveNamespaceFromEndpoint = 'true';
         tr.run();
```
Comment (ds-ms) [inline Tasks/KubernetesManifestV0/Tests/L0.ts]: Add a test for validating back compat as well

File: Tasks/AzureFunctionOnKubernetesV0/task.json [hunk 4]
```diff
@@ -15,7 +15,7 @@
     "version": {
```
Comment (ds-ms) [inline Tasks/AzureFunctionOnKubernetesV0/task.json]: You'll need to build these tasks as well, to update the `.loc.json` file or you can manually update them yourself

Comment (colindembovsky) [reply Tasks/AzureFunctionOnKubernetesV0/task.json]: @ds-ms not sure what you're asking here - I've updated the task.jsons for the affected tasks.

Comment (ds-ms) [reply Tasks/AzureFunctionOnKubernetesV0/task.json]: For every task.json there's a task.loc.json which gets updated when you build that task.
You're missing that update

**Comment by ds-ms** [comment]
Thanks for your contribution! :) @colindembovsky 

I kinda missed merging it a few times earlier post triggering the CI.

**Pull request merged**